src/util/map.ts

```diff
@@ -1,19 +1,19 @@
 export const TILE_SIZE = 256;
 export const MAP_SIZE = [24000, 20000];
 
-export type Point = [number, number];
+export type Point = [number, number, number];
 
-export function isValidXZ(x: number, z: number) {
-  return Math.abs(x) <= 6000 && Math.abs(z) <= 5000;
+export function isValidXYZ(x: number, y: number, z: number) {
+  return Math.abs(x) <= 6000 && Math.abs(z) <= 5000 && Math.abs(y) <= 2000;
```

Comment:
I don't think we should be checking the Y coordinate; Y doesn't really matter for map units.